x/stake/handler.go

```diff
@@ -184,31 +148,31 @@ func delegate(ctx sdk.Context, k Keeper, delegatorAddr sdk.Address,
 	if err != nil {
 		return nil, err
 	}
-	pool, candidate, newShares := pool.candidateAddTokens(candidate, bondAmt.Amount)
+	validator, pool, newShares := validator.addTokensFromDel(pool, bondAmt.Amount)
```

Comment:
whyd this move from pool  to validator

Reply:
it's much cleaner to organize this way, to me it's obvious that adding tokens from a delegator is a function of the validator not a function of the pool, it just so happens that the pool is needed to calculate exchange rates and is affected when a validator adds a delegated tokens, but first and foremost I see this as a feature which a validator has not one which the pool has which results in a new validator

Reply:
I'm a big fan of keeping the pool a low level object, functions on the pool affect only the pool